frontend/src/components/MyApplicationItem/MyMissionItem/useMission.ts

```diff
@@ -0,0 +1,112 @@
+import { useEffect, useMemo, useState } from "react";
+import { formatDateTime } from "../../../utils/format/date";
+import useMissionJudgment from "./useMissionJudgment";
+import { generatePath, useNavigate } from "react-router-dom";
+import useRefresh from "./useRefresh";
+import { Mission } from "../../../../types/domains/recruitments";
+import { PARAM, PATH } from "../../../constants/path";
+import { BUTTON_LABEL } from "../../../constants/recruitment";
+
+type MissionProps = {
+  mission: Mission;
+  recruitmentId: string;
+};
+
+const missionLabel = (submitted: boolean, missionStatus: Mission["status"]) => {
+  const labelMap = {
+    SUBMITTABLE: BUTTON_LABEL.BEFORE_SUBMIT,
+    SUBMITTING: submitted ? BUTTON_LABEL.EDIT : BUTTON_LABEL.SUBMIT,
+    UNSUBMITTABLE: BUTTON_LABEL.UNSUBMITTABLE,
+    ENDED: submitted ? BUTTON_LABEL.COMPLETE : BUTTON_LABEL.UNSUBMITTED,
+  } as const;
+
+  return labelMap[missionStatus];
+};
+
+const useMission = ({ mission, recruitmentId }: MissionProps) => {
+  const navigate = useNavigate();
+
+  const [missionItem, setMissionItem] = useState<Mission>({ ...mission });
+  const { isRefreshAvailable, fetchRefreshedResultData } = useRefresh({
+    missionItem,
+    recruitmentId: Number(recruitmentId),
+  });
+  const { fetchJudgmentMissionResult, isJudgmentAvailable } = useMissionJudgment({
+    missionItem,
+    recruitmentId: Number(recruitmentId),
+  });
+
+  const applyButtonLabel = missionLabel(mission.submitted, mission.status);
+
+  useEffect(() => {
+    setMissionItem(mission);
+  }, [mission]);
+
+  const formattedStartDateTime = useMemo(
+    () => (missionItem.startDateTime ? formatDateTime(new Date(missionItem.startDateTime)) : ""),
+    [missionItem.startDateTime]
+  );
+
+  const formattedEndDateTime = useMemo(
+    () => (missionItem.endDateTime ? formatDateTime(new Date(missionItem.endDateTime)) : ""),
+    [missionItem.endDateTime]
+  );
```

Comment:
c: `useMission` 훅이 여러 책임을 가지고 있는 것 같아요. Date에 대한 로직을 분리하면 테스트와 유지보수가 더 쉬워질 것 같아요~!

```suggestion
   const useMissionDates = (mission: Mission) => {
     return useMemo(() => ({
       formattedStartDateTime: mission.startDateTime ? formatDateTime(new Date(mission.startDateTime)) : "",
       formattedEndDateTime: mission.endDateTime ? formatDateTime(new Date(mission.endDateTime)) : "",
     }), [mission.startDateTime, mission.endDateTime]);
   };

   // useMission 내부
   const { formattedStartDateTime, formattedEndDateTime } = useMissionDates(missionItem);
```

Reply:
저도 훅이 다소 많은 기능을 가지고 있다는 점에 공감합니다☠️. 
그래서 이 부분은 향후 개선의 여지가 있다고 생각해요.

제안해 주신 코드는 날짜 포맷팅 로직을 별도의 훅으로 분리하는 것으로 이해했습니다. 너무 좋은 제안이지만🥲, 하지만 현재 상황에서 몇 가지 고려해야 할 부분이 있습니다.

1. `formattedStartDateTime`과 `formattedEndDateTime`은 현재 이 훅 내에서만 사용되는 상태입니다. 이들은 미션 컴포넌트에 필요한 정보를 제공하는 역할을 합니다.

2. 현재 구조에서도 `formatDateTime` 유틸 함수를 사용하여 날짜 포맷팅을 처리하고 있어, 이미 어느 정도 관심사 분리가 이루어져 있습니다.

3. 이 시점에서 추가적인 분리는 오버엔지니어링이 될 수 있다고 생각합니다. 코드의 복잡성이 증가할 수 있으며, 현재 구조로도 충분히 이해하기 쉽고 유지보수가 가능합니다.

다만, 앞으로 이 부분의 복잡성이 증가하거나 재사용성에 관해 추가적인 고려가 필요하다면, 준이 제안해 주신 방식대로 분리를 고려해 볼 수 있을 것 같아요~! 현재로서는 기능 요구사항을 충족하면서도 코드를 좀 더 단조로운 구조로 유지하는 것이 더 중요하다고 생각합니다.

향후 리팩토링 시 이 부분을 우선적으로 검토하도록 하겠습니다. 좋은 제안 감사드려요 😊